### PR TITLE
Experimental postgres plugin

### PR DESCRIPTION
**Pull request by matthewloring**

### CODE REVIEW

File: src/plugins/plugin-pg.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var shimmer = require('shimmer');
+
+var SUPPORTED_VERSIONS = '^6.x';
+
+function makeQueryWrap(api) {
```
Comment (kjin) [inline src/plugins/plugin-pg.js]: Suggestion: can we capture `api` in a closure here and in `wrapCallback`?

File: src/plugins/plugin-pg.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+function wrapCallback(api, span, done) {
+  var fn = function(err, res) {
+    if (api.enhancedDatabaseReportingEnabled()) {
+      if (err) {
+        span.addLabel('error', err);
+      }
+      if (res) {
+        // TODO(mattloring): summarize once it is available on the public api
```
Comment (kjin) [inline src/plugins/plugin-pg.js]: I think with recent changes this TODO can go away, right?

**Comment by kjin** [comment]
If it helps, you can apply this patch on top of this PR once you rebase: https://github.com/kjin/cloud-trace-nodejs/commit/dd50a45f44e07ea819bf15000aa884d9a0a95a2d

File: src/plugins/plugin-pg.js (hunk 2 above)
Comment (matthewloring) [inline src/plugins/plugin-pg.js]: Yup. removed.

File: src/plugins/plugin-pg.js (hunk 1 above)
Comment (matthewloring) [inline src/plugins/plugin-pg.js]: Simplified and moved into the closure.

**Comment by matthewloring** [comment]
Thank you very much for putting together the patch you linked! On cancelation, if the request has been buffered but not sent, then no async work has actually been done and there is no span to report. If the request has been sent and is canceled in flight then I think it should resolve normally with information in the response indicating that it was canceled (we should verify this though).

**Pull request merged**